tool/deploy_webapp.sh

```diff
@@ -1,17 +1,34 @@
 # Exit immediately if a command exits with a non-zero status
 set -e
-if [ $# -lt 2 ] || [ $# -gt 3 ]; then
-    echo "Usage: $0 [--skip-cloud-functions] path/to/firebase/deployment/constants.json path/to/crypto/token/file"
-    exit 1
-fi
 
 WORK_DIR="$(pwd)"
 
-########## validate argument(s)
-
-if [ "$1" = "--skip-cloud-functions" ]; then
-  SKIP_CLOUD_FUNCTIONS=true
-  shift
+########## parse and validate argument(s)
+
+FIREBASE_DEPLOY_ARGS=()
+
+while [ "$#" -gt 0 ]; do
+  case "$1" in
+    --skip-cloud-functions)
+      SKIP_CLOUD_FUNCTIONS=true
+      shift
+      ;;
+    --only)
+      shift
+      FIREBASE_DEPLOY_ARGS[0]="--only"
+      FIREBASE_DEPLOY_ARGS[1]="$1"
```

Comment:
Perhaps add `if [ -z "$1" ]; then...` here to validate that there is indeed an argument after the `--only`

Reply:
good idea, thanks @danrubel !

I've also added a `usage_and_exit` function as well and added it in the other places we were exiting with an error.